Dockerized variant of update.sh for project dependencies.

jmchilton opened the pull request.

jmchilton committed:
Dockerized variant of update.sh for project dependencies.

mvdbeek commented on lib/galaxy/dependencies/pipfiles/update.sh:
```diff
@@ -4,21 +4,26 @@ commit=0
 Use -c to automatically commit these changes (be sure you have no staged git
-changes).
+changes). Use -d to rebuild with Pipenv from the
+galaxy/update-python-dependencies container. This container can be built be
```
```suggestion
galaxy/update-python-dependencies container. This container can be built by
```

mvdbeek replied:
```suggestion
galaxy/update-python-dependencies container. This container can be built by
```

mvdbeek committed:
Apply suggestions from code review

Fix typo

Co-Authored-By: nsoranzo <nicola.soranzo@gmail.com>

nsoranzo committed:
Use ``python:2.7`` image

nsoranzo committed:
Fix environment marker for `typing`

Also follow-up to commit 340c2121b67e10440a6073e0da1a5256fd864fe5 : remove
`bunch` and fix `more-itertools` environment marker.

nsoranzo reviewed (approved):
@jmchilton I've rebased, added some fixes and changed the base Docker image to `python:2.7` (since I plan to play with adding more Dockerfiles for Python 3.5, 3.6 and 3.7), hope that's OK. This image is also slightly smaller and comes with `pip` preinstalled.

jmchilton commented:
Looks great to me, thanks!
